src/VisualStudio/Core/Def/Workspace/VisualStudioActiveDocumentTracker.cs

```diff
@@ -316,26 +316,16 @@ private void TryInitializeTextBuffer()
 
                 _documentTracker.AssertIsForeground();
 
-                if (ErrorHandler.Succeeded(Frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocCookie, out var boxedDocCookie)))
+                // NB: -5055 => __VSFPROPID12.VSFPROPID_IsDocDataInitialized available in 17.9
+                if (ErrorHandler.Succeeded(Frame.GetProperty((-5055), out var boxedIsDocDataInitialized)))
```

Comment:
can we update whatever reference we have so we can reference this by name, and not integer?

Reply:
Can we reference a preview version of the VS SDK? The NuGet Package Manager shows that 17.8 is the latest available version.
![image](https://github.com/dotnet/roslyn/assets/49529862/7b8adac8-98d3-4d0a-aa87-9b0f73b9dda0)

Reply:
We theoretically use any package version from this feed - https://dev.azure.com/azure-public/vside/_artifacts/feed/vssdk/NuGet/Microsoft.VisualStudio.Interop/versions/17.9.36400-preview.3 (which has 17.9 preview 3).

However it might cause our integration tests to fail because they run on a public preview (usually slightly behind the latest, for example we're on 17.8p6).  I think in a week or two the images will be upgraded to 17.9 P1).

If the property is only defined in 17.9p1 or above, I would be OK with having the number + a tracking bug (can assign to me) to switch to the real property when we're able to upgrade our integration tests.  Does that work for you @CyrusNajmabadi ?

Reply:
> I would be OK with having the number + a tracking bug (can assign to me) to switch to the real property when we're able to upgrade our integration tests.

+1 to this. I'd prefer getting this in sooner since it'll give us some more bake time for this change, help reduce some noise when CTI runs through our async open test plan and help reduce some noise in the UI delays that get reported.

Reply:
There are issues with 17.9 VSSDK that currently block us from upgrading. Working with the VS Extensibility team on addressing them.

https://github.com/dotnet/roslyn/pull/71111